Refine ECE SUSE host configuration page (#6598)
elastic/docs-content: deploy-manage/deploy/cloud-enterprise/configure-host-suse.md

Follow these steps to configure the user account according to the [Users and permissions prerequisites](ece-users-permissions.md) for ECE hosts.

1. Set up the OS groups and add your user.

    1. Create the `elastic` and `docker` groups if they don't already exist:

        ```sh
        sudo groupadd elastic
        sudo groupadd docker
        ```

    1. (Optional) Create a dedicated user for ECE.

        If the user you are currently logged in as is not the user that will install and run ECE, create a dedicated user. The following example creates the recommended `elastic` user:

        ```sh
        sudo useradd -m -g elastic -G docker elastic
        ```

        ::::{note}
        If you create a dedicated user, we recommend granting it `sudo` privileges and then logging in or switching to that user before continuing. This allows `$USER` to automatically resolve to the user that will install and run ECE.

        Alternatively, you can continue using a different account with `sudo` privileges, and replace `$USER` with the name of the user that will install and run ECE in the remaining commands in this guide.
        ::::

    1. Add the user to both groups:

        ```sh

>         sudo usermod -aG elastic,docker $USER

Reviewer: If the user decide to create `elastic` as $USER then the previous 
```bash
 sudo useradd -m -g elastic -G docker elastic
```
already sets the `elastic` user to have elastic,docker as groups.
What I am trying to say is that if the user takes the optional step, then this step could be skipped as these groups are already assigned.
This step is important if they decide to use an already existent user and we need to add (by appending) the needed groups.

Reviewer 2: Yeah, that step is only needed if the user didn't create a user in the previous step, good point. I'll fix it.

::::{note}
You can use LVM, `mdadm`, or a combination of the two for block device management. However, their configuration is outside the scope of this guide and is not covered by ECE support.
::::

::::{important}
You must use XFS and have quotas enabled on all allocators. Otherwise, disk usage won't display correctly.
::::

In the following example, XFS is set up on a single, pre-partitioned block device named `/dev/xvdg1`. Replace `/dev/xvdg1` with the corresponding device on your host.

1. Format the partition:

    ```sh
    sudo mkfs.xfs /dev/xvdg1
    ```

1. Add an entry to the `/etc/fstab` file for the new XFS volume. The default filesystem path used by ECE is `/mnt/data`:

    ```sh
    /dev/xvdg1	/mnt/data	xfs	defaults,pquota,prjquota,x-systemd.automount  0 0
    ```

## Prepare the data directories [ece-prepare-data-directories-sles]

Prepare the data directories used by {{ece}} and Docker. These steps create the `/mnt/data` and `/mnt/data/docker` directories, mount an XFS file system if applicable, and apply the required ownership and permissions.

1. Create the `/mnt/data` directory if it doesn't already exist:

    ```sh

>     sudo install -o $USER -g elastic -d -m 700 /mnt/data

Reviewer: TBH, since I didn't know for sure whether it actually did anything (apart from returning a 0 exit code), we may suggest to check that the newly created folder has elastic (owner) and elastic (group) as well as  drwx------ (700)
```
francescapesci@francesca-ece-suse:~> ls -ld /mnt/data
drwx------ 2 elastic elastic 6 Jul 14 16:13 /mnt/data
```

Reviewer: I see that https://github.com/elastic/docs-content/pull/6598/changes#diff-b13773bd6f750767482d21362e706bb0570cf173593672a4afcdfe7f1b3a8e0cR471 afterwards we test it as well, so this becomes a small nit 👍🏽

Reviewer 2: yeah, the verification is at the end of the doc. But it's a good point, I'll check this a bit further.

# Configure a SUSE host [ece-configure-hosts-sles12]

> SUSE Linux Enterprise Server (SLES) hosts use `zypper` to install Docker. They also require manual setup of XFS quotas, since SLES doesn't ship XFS as the default file system. The steps on this page target SLES 15. 

Reviewer 2: ```suggestion
SUSE Linux Enterprise Server (SLES) hosts use `zypper` to install Docker. They also require manual setup of XFS quotas, since SLES doesn't ship XFS as the default file system. The steps on this page target SLES 15 SP4. 
```

Not ideal but we have to clarify our steps are for SP4, as they don't work with SP7 today (SP7 doesn't come with a supported Docker version).

> Before installing, make sure to cross-check your SLES version and Docker version against the [Support matrix](https://www.elastic.co/support/matrix#elastic-cloud-enterprise). The commands shown on this page are examples. Substitute the versions you've identified in the support matrix.

Reviewer 3: ```suggestion
Before installing, make sure to cross-check the compatible SLES version and Docker version combination against the [Support matrix](https://www.elastic.co/support/matrix#elastic-cloud-enterprise). The commands shown on this page are examples. Substitute the versions you've identified in the support matrix.
```

Good to explicitly mention "compatible" so that user can understand more on "why" they need to cross check.

    sudo zypper update -y
    ```

1. List the available Docker versions:

    ```sh
    sudo zypper search -s -t package --match-exact docker
    ```

    Note the version you want to install. Check the [Support matrix](https://www.elastic.co/support/matrix#elastic-cloud-enterprise) for supported Docker versions for your OS. If the latest available version is compatible, you don't need to specify an explicit version in the next step.

1. Install Docker and other required packages on SLES 15. For example, to install Docker 25:

    ```sh
    sudo zypper install -y curl device-mapper lvm2 net-tools docker=25.0.6_ce-150000.207.1 <1>
    ```

    1. Replace `25.0.6_ce-150000.207.1` with the exact package version from the repository listing. Note that zypper does not support wildcards (for example, `docker=25.*`). To install the latest available version, specify `docker` without a version number.

    ::::{tip}
    If `zypper` reports that the requested Docker version isn't available, make sure the SUSE **Containers Module** is enabled. Refer to the [SUSE documentation](https://documentation.suse.com/) for instructions on adding the upstream Docker repository.
    ::::

> 1. Set up the OS groups and add your user.

Reviewer 2: ```suggestion
## Prepare the user account for ECE

The following commands assume that you are logged in as the non-root user account that will run ECE. We recommend using a dedicated `elastic` user, but you can also use an existing non-root user account with a UID greater than 1000.

1. Set up the OS groups and add your user.
```

Objectives of the suggestion:
- Separate user & group setup to a dedicated section.
- Tell (and recommend) readers that they can create a dedicated user account (i.e `elastic`) for ECE installation.

    1. Create the `elastic` and `docker` groups if they don't already exist:

        ```sh
        sudo groupadd elastic
        sudo groupadd docker
        ```

    1. Add the user to both groups:

        ```sh
        sudo usermod -aG elastic,docker $USER
        ```

    1. The user running ECE must have a UID and GID of at least 1000, and the primary GID must be set to the `elastic` group. To find the `elastic` group GID:

        ```sh
        grep elastic /etc/group
        ```

        Then set the user's primary group to `elastic`:

        ```sh
        sudo usermod -g <elastic_group_gid> $USER
        ```

> 1. Stop the `nscd` service and prevent it from starting automatically (it can interfere with Elastic services):

Reviewer 2: Let's move this item (nscd) somewhere else:

I'd suggest to move it to the "Update the configurations settings" / "System configuration" section.
Or place it at the end of "docker install" section, but not together with user & group configuration.

# Configure a SUSE host [ece-configure-hosts-sles12]

This guide explains how to prepare a SUSE Linux Enterprise Server (SLES) host for an {{ece}} (ECE) installation. It covers the operating system configuration required before you install ECE, including Docker installation, XFS quota configuration, and other host-specific settings.

SLES hosts use `zypper` to install Docker and require manual XFS quota setup because XFS is not the default filesystem on SLES. The steps on this page target SLES 15 SP4.

::::{warning}
SLES 12 SP5 reached general support end of life on **October 31, 2024**. Use SLES 15 or later for new {{ece}} installations, and migrate existing SLES 12 SP5 hosts.
::::

> * [Prerequisites](#ece-configure-host-suse-prerequisites)

Reviewer 4: this page nav duplicates the right hand nav. do we expect people will be jumping around this guide much? if we expect people to have to do all of these steps, I would skip putting this full list in

if the steps are order-sensitive (or partially order-sensitive) we should make sure that the steps get numbered or dependencies are called out in the prose (the latter for the 'only sometimes the order matters' case)

Reviewer 2: I've added a little extra prose to the relevant sections, the majority of them are independent.
Page nav removed. I agree it's better like this.

1. Set up the OS groups and add your user.

    1. Create the `elastic` and `docker` groups if they don't already exist:

        ```sh
        sudo groupadd elastic
        sudo groupadd docker
        ```

    1. (Optional) Create a dedicated user for ECE:

        If the user you are currently logged in as is not the user that will install and run ECE, create a dedicated user. The following example creates the recommended `elastic` user:

        ```sh
        sudo useradd -m -g elastic -G docker elastic
        ```

        ::::{note}
        If you create a dedicated user, we recommend granting it `sudo` privileges and then logging in or switching to that user before continuing. This allows `$USER` to automatically resolve to the user that will install and run ECE.

        Alternatively, you can continue using a different account with `sudo` privileges, and replace `$USER` with the name of the user that will install and run ECE in the remaining commands in this guide.
        ::::

    1. Add the user to both groups:

        ```sh
        sudo usermod -aG elastic,docker $USER
        ```

> 1. Verify that the user that will run ECE has a UID and GID of at least 1000:

Reviewer 4: mostly for my own info: why do we need this?

Reviewer 3: @eedugon @shainaraskas 

I didn't know why we removed the `4. Disable nscd, as it interferes with Elastic’s services:` but we do have a page for describing that. 
https://www.elastic.co/docs/deploy-manage/deploy/cloud-enterprise/ece-sysconfig

Maybe we shouldn't delete that line?

Reviewer 4: I think it's covered elsewhere:

https://github.com/elastic/docs-content/pull/6598#discussion_r3524158147

Reviewer 2: Exactly. It was moved to a sub-section that fits much better with the nature of the step.

Reviewer 2: @shainaraskas : 
> Verify that the user that will run ECE has a UID and GID of at least 1000:

If you are asking about that, it's an ECE limitation. But I don't remember exactly why, probably an ECE internal check to ensure it doesn't run as root 🤷

Reviewer 3: We have https://www.elastic.co/docs/deploy-manage/deploy/cloud-enterprise/ece-users-permissions mentioning this: 
> To install ECE: A user with a UID and GID greater than or equal to 1000 who is part of the docker group. You must not install ECE as the root user.

And IMHO, yes, that's more a best practice to have a non-root user (UID/GID >= 1000) on Linux distributions.

Reviewer 2: Besides the best practice (2-999 = other system or services users, 1000+ = other users) I'd swear we have something hardcoded in ECE that requires this "practice", and that's the reason of the `must`, but maybe I'm wrong.... anyway I wouldn't change it :)

    ```sh
    id $USER
    ```

    The output should show a `uid` and `gid` value of `1000` or higher.

1. Verify that the user's primary group is `elastic`:

    ```sh
    id -gn $USER
    ```

    If the command doesn't return `elastic`, find the `elastic` group GID:

    ```sh
    grep elastic /etc/group
    ```

    Then set the user's primary group to `elastic`:

    ```sh
    sudo usermod -g <elastic_group_gid> $USER
    ```

> ## Install Docker on SLES [ece-install-docker-sles12]

Reviewer 4: need an intro sentence after the heading

::::{include} /deploy-manage/deploy/_snippets/ece-supported-combinations.md
::::

> 1. Remove Docker and any previously installed Podman packages.

Reviewer 4: there is some inconsistency with punctuation use throughout (or as far as I see so far :)). generally I think we need to intro our code snippets with colons

```suggestion
1. Remove Docker and any previously installed Podman packages:
```

## Set up XFS quotas [ece-xfs-setup-sles12]

{{ece}} relies on XFS project quotas to manage disk space for {{es}} data directories. These quotas limit the amount of disk space available to each {{es}} cluster node based on the RAM-to-disk ratio defined by its [instance configuration](deployment-templates.md#ece-getting-started-instance-configurations). For example, the default `data.default` instance configuration uses a 1:32 ratio, allowing 32 GB of disk space for every 1 GB of RAM assigned to a cluster node.

To use disk quotas, the file system mounted at `/mnt/data` must be an XFS file system with project quotas enabled. This guide creates a dedicated XFS file system for `/mnt/data`, which is the recommended configuration. If `/mnt/data` already resides on an XFS file system with project quotas enabled, you can skip the file system creation steps.

::::{note}
You can use LVM, `mdadm`, or a combination of the two for block device management. However, their configuration is outside the scope of this guide and is not covered by ECE support.
::::

::::{important}
You must use XFS and have quotas enabled on all allocators. Otherwise, disk usage won't display correctly.
::::

> **Example:** Set up XFS on a single, pre-partitioned block device named `/dev/xvdg1`. Replace `/dev/xvdg1` in the following example with the corresponding device on your host.

Reviewer 4: would try to kill this `example` syntax. use something like "For example," or "In the following example .."

Before installing, make sure to cross-check your SLES version and Docker version against the [Support matrix](https://www.elastic.co/support/matrix#elastic-cloud-enterprise). The commands shown on this page are examples; substitute the versions you've identified in the support matrix.

::::{warning}
SLES 12 SP5 reached general support end of life on **October 31, 2024**. Use SLES 15 or later for new {{ece}} installations, and migrate existing SLES 12 SP5 hosts.
::::

* [Install Docker](#ece-install-docker-sles12)
* [Set up XFS quotas](#ece-xfs-setup-sles12)
* [Update the configurations settings](#ece-update-config-sles)
* [Configure the Docker daemon options](#ece-configure-docker-daemon-sles12)

## Install Docker on SLES [ece-install-docker-sles12]

::::{include} /deploy-manage/deploy/_snippets/ece-supported-combinations.md
::::

1. Remove Docker and any previously installed podman packages.

    ```sh
    sudo zypper remove -y docker docker-ce podman podman-remote
    ```

> 2. Update packages to the latest available versions.

Reviewer 2: ```suggestion
1. Update packages to the latest available versions.
```

Let's start all numbered list items with 1., for easier maintenance.

Author: fixed in 32578c3082ab381e0b7da516cac6d63ae93d61d1

    ```sh
    sudo zypper refresh
    sudo zypper update -y
    ```

> 3. Install Docker and other required packages on SLES 15. The following command is an example of installing Docker {{ece-docker-version}}. To install a different Docker version, replace {{ece-docker-version}} with your preferred version from the [Support matrix](https://www.elastic.co/support/matrix#elastic-cloud-enterprise).

Reviewer 2: ````suggestion
1. List the available versions of docker package in the repositories:

    ```sh
    suso zypper search -s -t package --match-exact docker
    ```

    From the output list make note of the version you want to install. Check the [Support matrix](https://www.elastic.co/support/matrix#elastic-cloud-enterprise) for the list of supported versions for your OS. If the latest available version is compatible, you can ignore the explicit version set in the following command.

1. Install Docker and other required packages on SLES 15. The following command is an example of installing Docker 25:
````

Author: fixed in 32578c3082ab381e0b7da516cac6d63ae93d61d1

    ```sh subs=true

>     sudo zypper install -y curl device-mapper lvm2 net-tools docker={{ece-docker-version}}.*

Reviewer 2: ```suggestion
    sudo zypper install -y curl device-mapper lvm2 net-tools docker=25.0.6_ce-150000.207.1 <1>
```

Reviewer 2: Note that `docker=27.0*` (or another type of wildcard) doesn't work with zypper. The version needs to match exactly the package version to install.

I've included `docker=25.0.6_ce-150000.207.1`, as Docker 25 is the officially supported package for SLES15 based on the support matrix `today`.

Author: fixed in 32578c3082ab381e0b7da516cac6d63ae93d61d1

>     ```

Reviewer 2: ````suggestion
    ```
    1. Replace `27.5.1_ce-150000.218.1` with the exact version to install, or just leave `docker` to install the latest available version of the docker package.
````

Author: fixed in 32578c3082ab381e0b7da516cac6d63ae93d61d1

    ::::{tip}

>     If `zypper` reports that the requested Docker version isn't available, make sure the SUSE **Containers Module** is enabled, or refer to [SUSE's documentation](https://documentation.suse.com/sles/15-SP6/html/SLES-all/cha-docker-installation.html) for adding the upstream Docker repository.

Reviewer 2: That link looks broken

    ::::

    ::::{note}
    Installation on SLES 12 SP5 is no longer covered here because SLES 12 SP5 is past general support end of life. If you're maintaining an existing SLES 12 SP5 deployment, install the last Docker version that SUSE shipped for SLES 12 SP5 and plan a migration to SLES 15.
    ::::

> 4. Set up the OS groups and add your user.

Reviewer 2: Here we have a signifficant deviation in the Ubuntu, RHEL and SuSE guides (cc: @kunisen )

Originally we suggested to create a user called `elastic` to install ECE. It looks at some point we allowed to install ECE with a different user. Some guides (RHEL) still show the `elastic` user, while the ubuntu guide uses the current user but then it shows content (/etc/limits.conf) with the `elastic` user hardcoded).

I think we should try to align the 3 installation guides, and if we support both ways we should state it clearly, and probably suggest the creation of the `elastic` user with an optional step.

Update: I can confirm that we support a normal (not `elastic`) user, but anyway we should align the 3 guides and probably include a section to do specific user and group setup.

Reviewer 2: For example @marciw , this TIP is great on the system limits:

<img width="797" height="609" alt="image" src="https://github.com/user-attachments/assets/5115fdc3-9f1b-4462-ae09-cb0ebc7ca090" />

And for that TIP to make better sense, when we talk about the user and groups we should also give the option of creating the `elastic` user, and now that's missing.

        sudo groupadd docker
        ```

    1. (Optional) If you want to use the recommended `elastic` user and it doesn't exist yet, create it:

        ```sh
        sudo useradd -m -g elastic -G docker elastic
        ```

>         The user that prepares the host must be able to run commands with `sudo`. Configure sudo access according to your organization's policies. For example, to allow the `elastic` user to run the commands in this guide without entering a password:

Reviewer 3: I have some questions on this:
* Should we mention this earlier like under the first `Prepare the user account for ECE` block?
* Do we have to write it in such a detailed way? (I guess no because I saw you removed this by looking at the latest status - https://github.com/elastic/docs-content/pull/6598/changes)

Reviewer 2: I've removed all that part if I'm not mistaken. And I agreed, that level of detail shouldn't be needed. Our doc's main focus should be ECE preparations and requirements, not general knowledge like how to configure sudo.

    ```

## Prepare the data directories [ece-prepare-data-directories-sles]

Prepare the data directories used by {{ece}} and Docker. These steps create the `/mnt/data` and `/mnt/data/docker` directories, mount an XFS file system if applicable, and apply the required ownership and permissions.

1. Create the `/mnt/data` directory if it doesn't already exist:

    ```sh
    sudo install -o $USER -g elastic -d -m 700 /mnt/data
    ```

1. If you configured a dedicated XFS file system in [Set up XFS quotas](#ece-xfs-setup-sles12):

    1. Mount the file system configured in `/etc/fstab`:

        ```sh
        sudo mount -a
        ```

    1. Set the ownership and permissions for `/mnt/data`:

        Mounting the XFS file system for the first time replaces the original mount point. Reapply the required ownership and permissions to the mounted file system.

        ```sh
        sudo chown $USER:elastic /mnt/data
        sudo chmod 700 /mnt/data
        ```

1. Create the `/mnt/data/docker` directory for Docker storage:

    ```sh
    sudo install -o $USER -g elastic -d -m 700 /mnt/data/docker
    ```

## Configure system settings [ece-update-config-sles]

> 1. Stop the `nscd` service and prevent it from starting automatically (it can interfere with Elastic services):

Reviewer 4: ```suggestion
1. Stop the `nscd` service and prevent it from starting automatically. This service can interfere with Elastic services.
```

Reviewer 3: maybe this is related to https://github.com/elastic/docs-content/pull/6598#discussion_r3524137890?

    ```sh
    sudo systemctl stop nscd
    sudo systemctl disable nscd
    ```

> 1. Enable cgroup accounting for memory and swap space.

Reviewer 4: when there is a sub-procedure we often also want colons

```suggestion
1. Enable cgroup accounting for memory and swap space:
```

    1. In the `/etc/default/grub` file, ensure the `GRUB_CMDLINE_LINUX=` variable includes these values:

        ```sh

>         cgroup_enable=memory swapaccount=1 cgroup.memory=nokmem

Reviewer 4: are these copypasteable, or is this more like a list of things to check? if the latter, consider wrapping thme each onto their own line

Reviewer 2: It's copypasteable! Better to leave it like that as the reader has to inject that entire string with spaces into the variable definition within the file.

We could offer an example of how the variable could look like.

        ```

    1. Update your Grub configuration:

        ```sh
        sudo update-bootloader
        ```

1. Configure kernel parameters:

    ```sh

>     cat <<EOF | sudo tee -a /etc/sysctl.conf

Reviewer 4: can we swap to a modern docs link here? can we swap these with code annotations?

Reviewer 2: solved ;)

    }
    ```

1. If you did not create the mount point earlier (if you did not set up XFS), create the `/mnt/data/` directory as a mount point:

    ```sh
    sudo install -o $USER -g elastic -d -m 700 /mnt/data
    ```

1. If you [set up a new device with XFS](#ece-xfs-setup-sles12) earlier:

    1. Mount the block device (change the device name if you use a different device than `/dev/xvdg1`):

        ```sh
        sudo mount /dev/xvdg1
        ```

    1. Set the permissions on the newly mounted device:

        ```sh
        sudo chown $USER:elastic /mnt/data
        ```

1. Create the `/mnt/data/docker` directory for Docker storage:

    ```sh
    sudo install -o $USER -g elastic -d -m 700 /mnt/data/docker
    ```

## Configure the Docker daemon [ece-configure-docker-daemon-sles12]

1. Edit `/etc/docker/daemon.json`, and make sure the following configuration values are present:

    ```json
    {
      "default-ulimits": {
        "nofile": {
          "Name": "nofile",
          "Soft": 1048576,
          "Hard": 1048576
        }

>       },

Reviewer 2: ```suggestion
```

default-ulimits -> nofile are already present in line 273.

    sudo install -o $USER -g elastic -d -m 700 /mnt/data/docker
    ```

> 

Reviewer 2: The step 7
> 7. If you [set up a new device with XFS](#ece-xfs-setup-sles12) earlier:

Is a bit contradictory, becuase if that was really done, both the mount and chown shouldn't be needed.

It feels like a mess, much better to just tell the user to ensure /mnt/data exist and with proper owner, group owner and permissions, and in case it's NOT correct, refer to the proper step, but write the config steps in a single location, and not twice.

Note that step 8 (/mnt/data/docker) is really needed.

Reviewer 2: In a similar approach as with user & group, probably creating the mount point & directories and setting permissions should be all together in its own separate section.

Author: Fixed in 32578c3082ab381e0b7da516cac6d63ae93d61d1

> 1. Edit `/etc/docker/daemon.json`, and make sure that the following configuration values are present:<br>

Reviewer 2: This `/etc/docker/daemon.json` file is missing critical info we added recently on the ubuntu file:

<img width="760" height="411" alt="image" src="https://github.com/user-attachments/assets/8ce5747f-24fb-4a78-99ab-57645380b7eb" />

In this case we need to simply combine all the content. The current content is valid and correct also, but we need to add those limits to the docker daemon.

Author: fixed in 32578c3082ab381e0b7da516cac6d63ae93d61d1

    ```

    ::::{important}
    The `net.ipv4.tcp_retries2` setting applies to all TCP connections and also affects the reliability of communication with systems other than {{es}} clusters. If your clusters communicate with external systems over a low quality network, you might need to select a higher value for `net.ipv4.tcp_retries2`.
    ::::

    1. Apply the settings:

        ```sh
        sudo sysctl -p
        ```

> 1. Adjust the system limits by adding the following configuration values to the `/etc/security/limits.conf` file. These values are based on the {{ecloud}} hosted offering and should be used for ECE as well. If needed, make sure to replace `elastic` with your user name.

Reviewer 4: maybe the fact that this is based on ech is not so important anymore as the two platforms are less related. but we can skip this "hosted offering" language

```suggestion
1. Adjust the system limits by adding the following configuration values to the `/etc/security/limits.conf` file. These values are based on {{ech}} and should be used for ECE as well. If needed, make sure to replace `elastic` with your user name.
```

Reviewer 3: ++ we may also change the ECH value too and also there's no absolute connection to say "we use this on ECH thus we use the same on ECE". 

> two platforms are less related.

But IMHO they are still related - including the backend and UI. 
It's just the users who use ECE may not be using ECH at the same time because they are on different Elastic subscriptions, so it doesn't make an absolute logic sense or it doesn't make an absolute need to explain the way like this. 

From a user perspective, I would think it would be better to explain in the following perspective:
* It makes system more stable and reliable, and it is designed to handle traffic load for production in a nice way, etc.

But that's a PM call to say it in that way or not, so if we are unsure what to call it out, ++ on @shainaraskas's suggestion to remove it at all.

Reviewer 2: I'm totally opened for any option here. As we already know, our OS guides are not aligned at all:

- RHEL : Doesn't mention anything about from where these `limits` come from <-- I'll follow this approach because that sentence doesn't add much benefit for a config file like this.

- Ubuntu and SuSE: `These values are derived from our experience with the Elastic Cloud hosted offering and should be used for Elastic Cloud Enterprise as well.`

I'm going to leave this shorter for the moment, aligned with the RHEL doc. Ubuntu doc will still have this longer sentence. And when we do the final alignment of all OS docs with devs and PMs we will decide the best way to present the `limits.conf` configuration.

@kunisen , I wouldn't add any prose to `sell` (stability & reliability) the values of this file. It's very common in the industry to just specify the limits and kernel parameters required by the applications without needing to explain all of them. And in this case `limits.conf` could not be as important as we think, as it doesn't apply to containers, only to pam-based sessions, and in ECE 95% of things run as containers.

So, IMO better to leave this simply as it is or without the ECH statement.

    }
    ```

> 2. The user installing ECE must have a User ID (UID) and Group ID (GID) of 1000 or higher. Make sure that the GID matches the ID of the `elastic` group created earlier (likely to be 1000). You can set this using the following command:

Reviewer 2: That step at this stage causes confusion, we should move it upwards, together with the creation of the user and group.

My suggestion is to have a single location for user, group and permissions setup.

The `usermod` mentioned here is probably needed because we need the user to have as `primary group`  the `elastic` group, but let's move it to the right place.

Another thing to add is a hint for the reader to get the GID of the `elastic` group. They could do that with `sudo cat /etc/group | grep elastic` for example (maybe there's a smoother way).

Author: fixed in 32578c3082ab381e0b7da516cac6d63ae93d61d1

    root             soft    memlock        unlimited
    ```

> 1. (Optional) Tune additional network kernel parameters for production workloads. Create a `70-cloudenterprise.conf` file in `/etc/sysctl.d/` and include these settings:

Reviewer 4: ```suggestion
1. Optional: Tune additional network kernel parameters for production workloads. Create a `70-cloudenterprise.conf` file in `/etc/sysctl.d/` and include these settings:
```

    ```sh
    cat << SETTINGS | sudo tee /etc/sysctl.d/70-cloudenterprise.conf
    net.ipv4.tcp_max_syn_backlog=65536
    net.core.somaxconn=32768
    net.core.netdev_max_backlog=32768
    net.ipv4.tcp_keepalive_time=1800
    net.netfilter.nf_conntrack_tcp_timeout_established=7200
    net.netfilter.nf_conntrack_max=262140
    SETTINGS
    ```

    Apply the settings:

    ```sh
    sudo sysctl --system
    ```

    :::{note}
    According to the [{{es}} networking settings](elasticsearch://reference/elasticsearch/configuration-reference/networking-settings.md), {{es}} overrides TCP keepalive settings at the socket level for its own connections:
    * If system-level values exceed 300 seconds, {{es}} automatically lowers them to 300 seconds.
    * Values below 300 seconds are used as is.

    For non-{{es}} connections such as the proxy layer, consider reducing the following TCP keepalive parameters to detect stale network sessions and prevent firewalls from dropping silent connections:
    * `net.ipv4.tcp_keepalive_time`
    * `net.ipv4.tcp_keepalive_intvl`
    * `net.ipv4.tcp_keepalive_probes`
    :::

> ## Configure the Docker daemon [ece-configure-docker-daemon-sles12]

Reviewer 4: haven't been commenting on this, but pref a little tl;dr of what is being done / the role of the thing you're configuring below each heading

Reviewer 2: included, together with the statement about data directories preparation and docker install steps should have been already performed.

> 1. Edit `/etc/docker/daemon.json`, and make sure the following configuration values are present:

Reviewer 4: ```suggestion
1. Edit `/etc/docker/daemon.json` to make sure the following configuration values are present:
```

    According to [{{es}} networking settings](elasticsearch://reference/elasticsearch/configuration-reference/networking-settings.md), {{es}} overrides TCP keepalive settings at the socket level for its own connections:
    * If system-level values exceed 300 seconds, {{es}} automatically lowers them to 300 seconds.
    * Values below 300 seconds are used as-is.

    For non-{{es}} connections such as the proxy layer, consider reducing the following TCP keepalive parameters to detect stale network sessions and prevent firewalls from dropping silent connections:
    * `net.ipv4.tcp_keepalive_time`
    * `net.ipv4.tcp_keepalive_intvl`
    * `net.ipv4.tcp_keepalive_probes`
    :::

>     1. Ensure settings in /etc/sysctl.d/*.conf are applied on boot:

Reviewer 2: The following script is broken, as `/var/lib/cloud/scripts/per-boot/00-load-sysctl-settings` cannot be created if the base directory doesn't exist.

We should double check with devs if that's still needed (check the other OS guides to see if we have that). It's a directory related with `cloud-init`, but not all OS images have cloud-init isntalled, and we don't mention it on our docs :(

For the command to work we could just do a `sudo mkdir -p  /var/lib/cloud/scripts/per-boot/`, but that wouldn't mean that the script would run on every boot.

Hopefully we should be able to remove all that part, find an alternative, or tell users that `cloud-init` is a pre-requisite for this installation.

Author: leaving this one open so we can get input from SMEs

        sudo systemctl enable docker
        ```

> 1. For best results, tune your network settings. Create a `70-cloudenterprise.conf` file in `/etc/sysctl.d/` and include these settings:

Reviewer 2: Tune network settings should belong to the previous section (Update the configuration settings or Update system configuration), as it's not related with Docker configuration.

Please address the other comments on these items before moving the content.

    ```

    :::{note}
    According to the [{{es}} networking settings](elasticsearch://reference/elasticsearch/configuration-reference/networking-settings.md), {{es}} overrides TCP keepalive settings at the socket level for its own connections:
    * If system-level values exceed 300 seconds, {{es}} automatically lowers them to 300 seconds.
    * Values below 300 seconds are used as is.

    For non-{{es}} connections such as the proxy layer, consider reducing the following TCP keepalive parameters to detect stale network sessions and prevent firewalls from dropping silent connections:
    * `net.ipv4.tcp_keepalive_time`
    * `net.ipv4.tcp_keepalive_intvl`
    * `net.ipv4.tcp_keepalive_probes`
    :::

>     1. Ensure the settings in `/etc/sysctl.d/*.conf` are applied at startup:

Reviewer 2: ```suggestion
    1. (Optional) Ensure `/etc/sysctl.d/*.conf` settings are applied at startup:
    
        This workaround requires `cloud-init` and is only needed if the settings are not automatically applied after a reboot.
```

      }
    }
    ```
    1. The `max-size` and `max-file` options configure rotation for the `json-file` logs created by each container. Adjust these values to match your logging requirements.
    2. The `default-ulimits` setting increases the maximum number of open file descriptors available to Docker containers.

> 1. _If the Docker registry doesn't require authentication, skip this step._

Reviewer 4: can we reword this to "if the docker registry requires authentication, then" or something?

        sudo systemctl restart docker
        ```

## Verify the host configuration [ece-verify-host-config-sles]

Reboot the host and verify that the required system, storage, and Docker configuration has been applied successfully.

1. Reboot your system to ensure that all configuration changes take effect:

    ```sh
    sudo reboot
    ```

>     Then log in again as your ECE user.

Reviewer 4: is this process obvious to the user?

Reviewer 2: Yes, how to log in as a certain user or become a certain user in the OS should be obvious to the reader.

Do you prefer to put that directly in the step?

> 1. Reboot your system and log in again as your ECE user to ensure that all configuration changes take effect:

Reviewer 4: no preference, just wondering :)

Reviewer 2: updated, hoping it's better now :)

    The output should show `/mnt/data` as an `xfs` file system with project quotas enabled.

1. Verify the ownership and permissions of the data directory:

    ```sh

>     stat -c "%U:%G %a %n" /mnt/data

Reviewer 3: @eedugon do we know why we want to drop this?

Reviewer 2: 1) We validate that docker root directory is `/mnt/data/docker` in a previous verification step.

2) Because `/mnt/data/docker` should belong to `root:root`, not to the `ECE user:elastic` :) So I didn't want the user to enter both directories and explain that the docker one should belong to root.

3) What we really care before installing ECE is that the user and group elastic are the owners of `/mnt/data` and with the right privileges. That would make `/mnt/data/elastic` to be created properly during the installation, so it's the main check IMO.

Anyway if you prefer we can also mention both directories in the command, but we should update the statement and tell the user that /mnt/data should belong to the ECE user but /mnt/data/docker should be `root:root`.
